src/test/java/org/commcare/formplayer/tests/CaseClaimTests.java

```diff
@@ -57,6 +57,66 @@ protected String getMockRestoreFileName() {
         return "restores/caseclaim.xml";
     }
 
+    @Test
+    public void testDefaultsOverride() throws Exception {
+        configureQueryMock();
+
+        // 1. No user inputs
+        Hashtable<String, String> inputs = new Hashtable<>();
+        QueryData queryData = new QueryData();
+        queryData.setInputs("search_command.m1", inputs);
+        queryData.setExecute("search_command.m1", true);
+        sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                "caseclaim",
+                queryData,
+                true,
+                EntityListResponse.class);
+
+
+        // 2. Override default with an input not in defaults
+        inputs.put("case_type", "case4");
+        sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                "caseclaim",
+                queryData,
+                true,
+                EntityListResponse.class);
+
+        // 3. Override default with an input that is present in defaults
+        inputs.put("case_type", "case2");
+        sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                "caseclaim",
+                queryData,
+                true,
+                EntityListResponse.class);
+
+        // 4. Override default with multiple inputs one valid and other invalid
+        inputs.put("case_type", "case2#,#case4");
+        sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                "caseclaim",
+                queryData,
+                true,
+                EntityListResponse.class);
+
+        // 5. Add an user input not present in defaults
+        inputs.put("name", "chris");
+        sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                "caseclaim",
+                queryData,
+                true,
+                EntityListResponse.class);
+
+        verify(webClientMock, times(3)).get(uriCaptor.capture());
+        List<URI> uris = uriCaptor.getAllValues();
+        // 1 and 2
+        assert uris.get(0).equals(new URI("http://localhost:8000/a/test/phone/search/?case_type=case1&case_type=case2&case_type=case3&include_closed=False"));
+
+        // 3 and 4
+        assert uris.get(1).equals(new URI("http://localhost:8000/a/test/phone/search/?case_type=case2&include_closed=False"));
+
+        // 5
+        assert uris.get(2).equals(new URI("http://localhost:8000/a/test/phone/search/?case_type=case2&name=chris&include_closed=False"));
```

Comment:
It would make the test easier to read if you put the asserts right after the relevant calls. Event better would be to split up the test in to 5 different tests, one for each case.

Reply:
makes sense https://github.com/dimagi/formplayer/pull/1003/commits/aff575805a1a34404ae5178e889cd845de1c1e26